src/js/hyperschedule.js

```diff
@@ -1669,21 +1669,66 @@ function downloadPDF()
           course.courseName,
           width - 12,
         );
+        const courseLocationLines = pdf.splitTextToSize(
+          slot.location,
+          width - 12,
+        );
+        const courseInstructorsLines = pdf.splitTextToSize(
+          course.faculty.map(fullName => fullName.split(",")[0]).join(", "),
+          width - 12
+        );
 
+        // Attributes to be shown are calculated based on space available
+        // order of preference: Code, location, name, professor
+        // order of display: code, name, professor, location
+        const maxTextLength = (yEnd - yStart) - 2 * pdf.getLineHeight();
+        const toLocationLength = courseCodeLines.length +
+              courseLocationLines.length;
+        const toNameLength = toLocationLength + courseNameLines.length;
+        const toInstructorLength = toNameLength + courseInstructorsLines.length;
+        const showLocation = toLocationLength * pdf.getLineHeight() <
+              maxTextLength + .5 * pdf.getLineHeight();
+        const showName = toNameLength  * pdf.getLineHeight() < maxTextLength;
+        const showInstructor = toInstructorLength  * pdf.getLineHeight() < maxTextLength;
+        const totalLength = Math.max(courseCodeLines.length,
```

Comment:
This whole construct of computing all the different possible lengths and then checking which are too long and then getting the longest one that isn't too long feels kinda gross, but I couldn't come up with a cleaner way to do it.

Reply:
How about making a list `["code", "location", "name", ...]` and then iterating through it and incrementing the `totalLength` until it gets too big? You can figure out how much to increment by on each iteration by looking up in a map `{ code: courseCodeLines.length, ... }`. Then by the end of the loop you know which things to add.